src/averageAges.js

```diff
@@ -54,11 +86,29 @@ function calculateWomenAverageAge(people, withChildren) {
  *
  * @return {number}
  */
+
+//   // 1. find a mother of each person (or only for men)
+//   // 2. keep people who have mothers in the array
+//   // 3. calculate the difference child.born - mother.born
+//   // 4. return the average value
+
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // 1. find a mother of each person (or only for men)
-  // 2. keep people who have mothers in the array
-  // 3. calculate the difference child.born - mother.born
-  // 4. return the average value
+  const ALL = onlyWithSon
+    ? people.filter(person => person.sex === 'm')
+    : people;
+
+  const AGES = ALL
+    .map(person => {
+      const MOTHER = people.find(woman => woman.name === person.mother);
+
+      return MOTHER ? person.born - MOTHER.born : null;
+    })
+    .filter(el => el);
+
+  const AVERAGE_AGE_DIFF = AGES
+    .reduce((a, b) => a + b) / AGES.length;
+
+  return Number(AVERAGE_AGE_DIFF.toFixed(2));
```

Comment:
typically we use uppercased variables for hardcoded values, these are calculated